src/fdb5/api/FDB.h

```diff
@@ -72,73 +76,200 @@ class FDB {
     FDB(FDB&&)            = default;
     FDB& operator=(FDB&&) = default;
 
-    // -------------- Primary API functions ----------------------------
+    // -------------- Primary API functions ----------------------------------------------------------------------------
 
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @param handle eckit::message::Message to data to archive
     void archive(eckit::message::Message msg);
+
+    /// Archive binary data to a FDB.
+    /// Reads messages from the eckit::DataHandle and calls archive() the the
+    /// corresponding messages.
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param handle eckit::DataHandle reference data to archive
```

Comment:
Archives a stream of potentially multiple messages.

Reply:
done